src/basics.ml

```diff
@@ -216,3 +216,53 @@ let term_of_rhs : rule -> term = fun r ->
     TE_Some(Bindlib.unbox (Bindlib.bind_mvar vars p))
   in
   Bindlib.msubst r.rhs (Array.mapi fn r.vars)
+
+(** {3 Subterm position helper module} *)
+
+(** Some tools to encode the position of a subterm in a term. *)
+module Subterm =
+struct
+  (** Each element of the list is a level in the tree of the term.  For
+      instance, in the term [Appl(Appl(f, x), Appl(Appl(g, a), b))], the
+      subterm [a] has position [1.0], encoded by [[0 ; 1]], [b] has [1.1]
+      encoded by [[1 ; 1]] and [x] has [0] encoded by [[0]]. *)
+  type t = int list
```

Comment:
Why putting positions in reverse order?

Reply:
To use a cons instead of an append when going to successive position

Reply:
It would be good to give an explanation here of why you think it is more efficient!

For instance, is compare (which uses 2 List.rev's) called often?

Reply:
Cons is more efficient than append, isn't it?  But it's true that List.rev has a cost

Reply:
I've introduced a more efficient comparison in commit 28ee74e